File: patterns/java/src/refactoring_guru/patterns/fasad/example/VideoFile.java
package refactoring_guru.patterns.fasad.example;

public class VideoFile {
    private String name;
    private String codecType;

    public VideoFile(String name) {
        this.name = name;
        this.codecType = name.substring(name.indexOf(".") + 1, name.length());
    }

    public String getCodecType() {
        return codecType;
    }

    public String getName() {
        return name;
    }
}

File: patterns/cs/src/RefactoringGuru/Facade/Example/SomeComplexMediaLibrary/VideoFile.cs
using System;

namespace RefactoringGuru.Facade.Example.SomeComplexMediaLibrary
{
    class VideoFile
    {
        private String name;
        private String codecType;

        public VideoFile(String name)
        {
            this.name = name;
            int index = name.IndexOf('.');
            int length = name.Length - (name.IndexOf('.') + 1);
            codecType = name.Substring(index, length);
        }

        public String GetCodecType()
        {
            return codecType;
        }

        public String GetName()
        {
            return name;
        }
    }
}
